### PR TITLE
Use upsampled dimensions in GetCurrentDimensions

### PR DESCRIPTION
**Pull request by eustas**
djxl_fuzzer expects that frame header layer_info contains dimensions
that correspond to what is passed to callback.

### CODE REVIEW

File: lib/jxl/decode_test.cc [hunk 1]
```diff
@@ -2986,135 +2986,144 @@ TEST(DecodeTest, SkipFrameWithAlphaBlendingTest) {
...
+      EXPECT_EQ(JXL_DEC_SUCCESS, JxlDecoderProcessInput(dec));
+      JxlThreadParallelRunnerDestroy(runner);
+      JxlDecoderDestroy(dec);
+    }
+
+    EXPECT_EQ(0u, jxl::test::ComparePixels(frames[0].data(), frames[4].data(),
+                                           oxsize, oysize, format, format));
+  };
+
+  for (bool keep_orientation : {true, false}) {
```
Comment (veluca93) [inline lib/jxl/decode_test.cc:3115]: Not necessary for now, but could we start converting some of those nested for loops into parameterized tests?

**Review by veluca93 — APPROVED** [review]

**Comment by jonsneyers** [comment]
This is the same fix as https://github.com/libjxl/libjxl/pull/1047, right?

File: lib/jxl/decode_test.cc (hunk 1 above)
Comment (eustas) [reply lib/jxl/decode_test.cc:3116]: Ack. Sure, make sense.

**Comment by eustas** [comment]
Yup, the same, plus some testing. At least it fails without the fix.

**Review by veluca93 — APPROVED** [review]

**Pull request merged**